src/main/org/nlogo/workspace/ModelsLibrary.java

```diff
@@ -200,14 +196,24 @@ private static void scanDirectory(java.io.File directory, Node grandparent, Node
               + (file.isDirectory() ? "/" : ""),
           file.isDirectory());
       if (child.isFolder()) {
+        hasChild = true;
         scanDirectory(file, parent, child, exclusive);
       } else {
         if (fileName.toUpperCase().endsWith(".NLOGO") ||
             fileName.toUpperCase().endsWith(".NLOGO3D")) {
+          hasChild = true;
           parent.add(child);
         }
       }
     }
+    // Only add nodes when they have at least one model or folder
+    // under them, and they're not the root `models` folder.
+    // See: https://github.com/NetLogo/models/issues/33
+    // (this would still allow a folder with just an empty folder under it, but
+    // I think we can be careful enough to avoid that situation) NP 2015-03-27.
+    if (hasChild && grandparent != null) {
+      grandparent.add(parent);
+    }
```

Comment:
Oh, yes, you're completely right! And that's a more likely situation (though it doesn't occur at the moment). So, yeah, maybe open an issue? You can even assign me. I'd love to rewrite the whole thing in functional style Scala and just recursively prune empty folders once the whole tree is constructed...